android/src/main/java/com/imagepicker/Utils.java

```diff
@@ -396,22 +395,65 @@ static String getMimeType(Uri uri, Context context) {
         return "Unknown";
     }
 
-    static @Nullable String getMimeTypeFromCursor(ContentResolver contentResolver, Uri uri) {
-        Cursor cursor = contentResolver.query(uri, null, null, null, null);
+    static @Nullable String getMimeTypeForContent(Uri uri, Context context) {
+        String fileName = getFileNameForContent(uri, context);
         String fileType = "Unknown";
+
+        int lastDotIndex = fileName.lastIndexOf('.');
+        if (lastDotIndex != -1) {
+            fileType = fileName.substring(lastDotIndex + 1);
+        }
+        return fileType;
+    }
+
+    static String getFileName(Uri uri, Context context) {
+        if (uri.getScheme().equals("file")) {
+            return uri.getLastPathSegment();
+        } else if (uri.getScheme().equals("content")) {
+            return getFileNameForContent(uri, context);
+        }
+
+        return "Unknown";
+    }
+
+    static String getOriginalFilePath(Uri uri, Context context) {
+        String originPath;
+        if (uri.getScheme().contains("content")) {
+            originPath = getFilePathFromContent(uri, context);
+            uri = getAppSpecificStorageUri(uri, context);
```

Comment:
This causes a second cached file to be created which isn't returned. Issue created here: https://github.com/react-native-image-picker/react-native-image-picker/issues/2271